src/printer.js

```diff
@@ -181,6 +181,20 @@ function genericPrintNoParens(path, options, print, args) {
     case "EmptyStatement":
       return "";
     case "ExpressionStatement":
+      // Detect Flow-parsed directives
+      if (n.directive) {
+        return concat([
+          nodeStr(
+            {
```

Comment:
Ok, I added https://github.com/josephfrazier/prettier/commit/c65cf2fe24d3c41cd92d4ee57fa8308317725e6a with what I think is the right idea here. Let me know what you think.

Note that I can't re-open the pull request, due to https://github.com/isaacs/github/issues/361. Once we get everything satisfactory, I can rebase and open a new PR with equivalent changes.